aten/src/THCS/generic/THCSTensor.c

```diff
@@ -176,62 +205,86 @@ THCSTensor *THCSTensor_(newWithTensorAndSize)(THCState *state, THCIndexTensor *i
   THCSTensor_(_move)(state, self, THCIndexTensor_(newWithTensor)(state, indices), THCTensor_(newWithTensor)(state, values));
 
   return self;
+#else
+  return NULL;
+#endif
 }
 
 THCSTensor *THCSTensor_(newWithSize)(THCState *state, THLongStorage *size)
 {
+#if !defined(__HIP_PLATFORM_HCC__)
   THCSTensor *self = THAlloc(sizeof(THCSTensor));
   THCSTensor_(rawInit)(state, self);
   THCSTensor_(rawResize)(state, self, size->size, 0, size->data);
 
   return self;
+#else
+  return NULL;
+#endif
 }
 
 THCSTensor *THCSTensor_(newWithSize1d)(THCState *state, int64_t size0)
 {
+#if !defined(__HIP_PLATFORM_HCC__)
   int64_t size[1] = {size0};
 
   THCSTensor *self = THAlloc(sizeof(THCSTensor));
   THCSTensor_(rawInit)(state, self);
   THCSTensor_(rawResize)(state, self, 1, 0, size);
 
   return self;
+#else
+  return NULL;
+#endif
 }
 
 THCSTensor *THCSTensor_(newWithSize2d)(THCState *state, int64_t size0, int64_t size1)
 {
+#if !defined(__HIP_PLATFORM_HCC__)
   int64_t size[2] = {size0, size1};
 
   THCSTensor *self = THAlloc(sizeof(THCSTensor));
   THCSTensor_(rawInit)(state, self);
   THCSTensor_(rawResize)(state, self, 2, 0, size);
 
   return self;
+#else
+  return NULL;
+#endif
 }
 
 THCSTensor *THCSTensor_(newWithSize3d)(THCState *state, int64_t size0, int64_t size1, int64_t size2)
 {
+#if !defined(__HIP_PLATFORM_HCC__)
   int64_t size[3] = {size0, size1, size2};
 
   THCSTensor *self = THAlloc(sizeof(THCSTensor));
   THCSTensor_(rawInit)(state, self);
   THCSTensor_(rawResize)(state, self, 3, 0, size);
 
   return self;
+#else
+  return NULL;
```

Comment:
can you change the #else conditions to `THError("not supported on HIP_PLATFORM")`

Reply:
Sure